units/src/locktime/relative.rs

```diff
@@ -152,3 +152,22 @@ impl fmt::Display for TimeOverflowError {
 
 #[cfg(feature = "std")]
 impl std::error::Error for TimeOverflowError {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn from_seconds_ceil_success() {
+        let actual = Time::from_seconds_ceil(100).unwrap();
+        let expected = Time(1_u16);
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn from_seconds_ceil_causes_time_overflow_error() {
+        let maximum_encodable_seconds = u16::MAX as u32 * 512;
+        let result = Time::from_seconds_ceil(maximum_encodable_seconds + 1);
+        assert!(result.is_err());
```

Comment:
Could you also check that constructing from `maximum_encodable_seconds` does **not** error?

Reply:
good call, added this test case as well